solarwinds_apm/apm_logging.py

```diff
@@ -38,13 +38,46 @@
 import os
 
 
+class ApmLoggingType:
+    """Mapping of solarwinds_apm library log types"""
+
+    log_types = {
+        "STDERR": 0,
+        "STDOUT": 1,
+        "FILE": 2,
+        "NULL": 3,
+        "DISABLED": 4,
+    }
+
+    @classmethod
+    def default_level(cls):
+        """Returns integer representation of default log type"""
+        return cls.log_types["STDERR"]
+
+    @classmethod
+    def disabled_level(cls):
+        """Returns integer representation of disabled log type"""
+        return cls.log_types["DISABLED"]
+
+    @classmethod
+    def is_valid_level(cls, level):
+        """Returns True if the provided level is a valid interger representation of log type, False otherwise."""
+        try:
+            level = int(level)
+            return bool(level in list(cls.log_types.values()))
+        except (ValueError, TypeError):
+            return False
+
+
 class ApmLoggingLevel:
-    """Abstract mapping class providing a conversion between solarwinds_apm agent logging level and Python logging module
+    """Abstract mapping class providing a conversion between solarwinds_apm library logging level and Python logging module
     logging levels.
     The solarwinds_apm package has seven different log levels, which are defined in the debug_levels dictionary.
     """
 
     # maps string representation of solarwinds_apm.sw_logging levels to their integer counterpart
+    # TODO no longer allow -1
+    # https://swicloud.atlassian.net/browse/NH-69517
     debug_levels = {
```

Comment:
I'm pretty sure the available levels and their meanings changed with oboe 14, not sure if you already updated these though

Reply:
We're going to keep level `-1` as a valid level for customers to config log disabling, though it shouldn't be used internally. If level is `-1` then log type will be disabled by `ApmConfig.update_log_type` and log level should be ignored by oboe.